### PR TITLE
WIP: virt-manager: enable aarch64-darwin

### PR DESCRIPTION
**Pull request by kloenk**
###### Motivation for this change
virt-manager on aarch64-darwin.

###### Things done
enabled many packages for `platforms.darwin`.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Kloenk** [comment]
This builds, but virt-manager is segfaulting, so not usable at this time.

**Comment by Kloenk** [comment]
It now works on x86_64-darwin (only can test it via rosetta)

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -17083,7 +17083,9 @@ in
 
   libversion = callPackage ../development/libraries/libversion { };
 
-  libvirt = callPackage ../development/libraries/libvirt { };
+  libvirt = callPackage ../development/libraries/libvirt {
+    inherit (darwin.apple_sdk.frameworks) Carbon AppKit;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:17087]:
```suggestion
    inherit (darwin.apple_sdk.frameworks) AppKit Carbon;
```

File: pkgs/tools/admin/gtk-vnc/default.nix [hunk 2]
```diff
@@ -49,9 +53,11 @@ stdenv.mkDerivation rec {
     glib
     libgcrypt
     cyrus_sasl
-    libpulseaudio
     gtk3
-  ];
+  ] ++ lib.optional stdenv.isLinux libpulseaudio;
+
+  mesonFlags = [
+  ] ++ lib.optional stdenv.isDarwin "-Dpulseaudio=disabled";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/gtk-vnc/default.nix:60]:
```suggestion
  mesonFlags = lib.optional stdenv.isDarwin "-Dpulseaudio=disabled";
```

File: pkgs/development/python-modules/pycairo/default.nix [hunk 3]
```diff
@@ -35,6 +35,7 @@ buildPythonPackage rec {
     cairo
   ];
 
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pycairo/default.nix:38]: Do the tests not work on aarch64-darwin?

File: pkgs/development/libraries/vte/default.nix [hunk 4]
```diff
@@ -87,6 +86,11 @@ stdenv.mkDerivation rec {
     };
   };
 
+  mesonFlags = [
+  ] ++ lib.optional stdenv.isDarwin [
+    "-D_b_symbolic_functions=false"
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/vte/default.nix:92]:
```suggestion
  mesonFlags = lib.optionals stdenv.isDarwin [
    "-D_b_symbolic_functions=false"
  ];
```

File: pkgs/development/libraries/libvirt/default.nix [hunk 5]
```diff
@@ -76,7 +78,7 @@ in stdenv.mkDerivation rec {
   ] ++ optionals enableGlusterfs [
     glusterfs
   ] ++ optionals stdenv.isDarwin [
-    libiconv gmp
+    libiconv gmp gnumake Carbon AppKit
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libvirt/default.nix:81]: Isn't gnumake part of the stdenv?

File: pkgs/development/libraries/libvirt/default.nix [hunk 6]
```diff
@@ -6,6 +6,8 @@
 , dnsmasq, libnl, libpcap, libxslt, xhtml1, numad, numactl, perlPackages
 , curl, libiconv, gmp, zfs, parted, bridge-utils, dmidecode, dbus, libtirpc, rpcsvc-proto, darwin
 , meson, ninja, audit, cmake, bash-completion, pkg-config
+, Carbon, AppKit
+, gnumake
 , enableXen ? false, xen ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libvirt/default.nix:11]:
```suggestion
, enableXen ? false, xen
```

File: pkgs/applications/virtualization/virt-manager/default.nix [hunk 7]
```diff
@@ -53,7 +53,8 @@ python3Packages.buildPythonApplication rec {
     gappsWrapperArgs+=(--prefix PATH : "${makeBinPath [ cpio e2fsprogs file findutils gzip ]}")
   '';
 
-  checkInputs = with python3Packages; [ cpio cdrtools pytestCheckHook ];
+  doCheck = false;
+  checkInputs = with python3Packages; [ cpio pytestCheckHook ] ++ lib.optional stdenv.isLinux cdrtool;
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-manager/default.nix:57]:
```suggestion
  checkInputs = with python3Packages; [ cpio pytestCheckHook ]
    ++ lib.optional stdenv.isLinux cdrtool;
```

File: pkgs/applications/virtualization/virt-manager/default.nix [hunk 8]
```diff
@@ -53,7 +53,8 @@ python3Packages.buildPythonApplication rec {
     gappsWrapperArgs+=(--prefix PATH : "${makeBinPath [ cpio e2fsprogs file findutils gzip ]}")
   '';
 
-  checkInputs = with python3Packages; [ cpio cdrtools pytestCheckHook ];
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-manager/default.nix:56]: Do the tests not work on aarch64-darwin?

**Review by SuperSandro2000 — COMMENTED** [review]
Please convert all ``platforms.linux ++ platforms.darwin`` to ``platforms.unix``

File: pkgs/applications/virtualization/virt-manager/default.nix [hunk 9]
```diff
@@ -33,8 +33,8 @@ python3Packages.buildPythonApplication rec {
   ] ++ optional spiceSupport spice-gtk;
 
   propagatedBuildInputs = with python3Packages; [
-    pygobject3 ipaddress libvirt libxml2 requests cdrtools
-  ];
+    pygobject3 ipaddress libvirt libxml2 requests 
+  ] ++ lib.optional stdenv.isLinux cdrtool;
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-manager/default.nix:37]:
```suggestion
  ] ++ lib.optional stdenv.isLinux cdrtools;
```

File: pkgs/applications/virtualization/virt-manager/default.nix [hunk 10]
```diff
@@ -33,8 +33,8 @@ python3Packages.buildPythonApplication rec {
   ] ++ optional spiceSupport spice-gtk;
 
   propagatedBuildInputs = with python3Packages; [
-    pygobject3 ipaddress libvirt libxml2 requests cdrtools
-  ];
+    pygobject3 ipaddress libvirt libxml2 requests
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virt-manager/default.nix:36]:
```suggestion
    pygobject3 ipaddress libvirt libxml2 requests
```

File: pkgs/development/libraries/spice-gtk/default.nix [hunk 11]
```diff
@@ -89,33 +92,36 @@ stdenv.mkDerivation rec {
...
   PKG_CONFIG_POLKIT_GOBJECT_1_POLICYDIR = "${placeholder "out"}/share/polkit-1/actions";
 
+
   mesonFlags = [
     "-Dcelt051=disabled"
     "-Dpulse=disabled" # is deprecated upstream
+    "-Dpolkit=${if withPolkit then "enabled" else "disabled"}"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "-Dwebdav=disabled"
+    "-Dusbredir=disabled"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/spice-gtk/default.nix:124]:
```suggestion
    "-Dusbredir=disabled"
```

File: pkgs/development/python-modules/pycairo/default.nix (hunk 3 above)
Comment (Kloenk) [reply pkgs/development/python-modules/pycairo/default.nix:38]: No, they failed. Would like to der able them. But did not figured out how yet

File: pkgs/development/libraries/libvirt/default.nix (hunk 5 above)
Comment (Kloenk) [reply pkgs/development/libraries/libvirt/default.nix:81]: Could be. libvirt expects gnumake as gmake. Added this as debugging.

**Comment by AndersonTorres** [comment]
Closing because https://github.com/NixOS/nixpkgs/pull/152350 will supercede this.

**Pull request closed without merging**